src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -169,7 +166,10 @@ public boolean checkInWareHouse(@NotNull final AbstractBuilding buildingEntry, b
                     if(addToList)
                     {
                         buildingEntry.setOnGoingDelivery(true);
-                        list.add(buildingEntry);
+                        if (!list.contains(buildingEntry))
+                        {
+                            list.add(buildingEntry);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "add()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)